src/Adapter/MSTest.CoreAdapter/Discovery/UnitTestDiscoverer.cs

```diff
@@ -75,63 +76,69 @@ internal UnitTestDiscoverer()
                 testElements.Count,
                 source);
 
-            this.SendTestCases(source, testElements, discoverySink);
+            this.SendTestCases(source, testElements, discoverySink, runSettings);
         }
 
-        internal void SendTestCases(string source, IEnumerable<UnitTestElement> testElements, ITestCaseDiscoverySink discoverySink)
+        internal void SendTestCases(string source, IEnumerable<UnitTestElement> testElements, ITestCaseDiscoverySink discoverySink, IRunSettings runSettings)
         {
+            var isDesignMode = RunSettingsUtilities.IsDesignMode(runSettings?.SettingsXml);
```

Comment:
How is the code related to `isDesignMode` going to get modified when collect source information is supported?

Is it a good idea to compute the business logic of `DesignMode` and `CollectSourceInformation` in this unit?